virttest/utils_misc.py

```diff
@@ -4558,3 +4558,39 @@ def get_pid(name, session=None):
         return None
     else:
         return int(output.split()[1])
+
+
+def get_sosreport(path=None, remote_ip=None, remote_pwd=None, remote_user="root"):
+    """
+    Get sosreport in host/guest
+
+    :param path: local host path for sosreport to be saved, defaults to logdir
+    :param remote_ip: remote host/guest ip address
+    :param remote_pwd: remote host/guest password
+    :param remote user: remote host/guest username
+
+    :return: sosreport log path on success, None on fail
+    """
+    from avocado.core import data_dir
+    from virttest import remote
```

Comment:
Actually moving that top would have cyclic import issue, so I thought it would be specific to this API and let us keep it local. Also I felt having this API in utils_misc.py is more appropriate than any other file.